specifyweb/frontend/js_src/lib/components/WbPlanView/helpers.ts

```diff
@@ -81,7 +83,7 @@ export async function savePlan({
       expectedResponseCodes: [Http.NO_CONTENT],
     }
   ).then(async () =>
-    newlyAddedHeaders.length === 0
+    newlyAddedHeaders.length === 0 && !isModified
       ? Promise.resolve()
```

Comment:
code is going to be more complicated than this - rather than simply removing the column from the dataset, you also have to delete the data that was in that column (probably that has to be done on the back-end)